src/ResourceLoader/ResourceLoaderArticlesModule.php

```diff
@@ -39,6 +45,55 @@ protected function getPages( ResourceLoaderContext $context ) {
 		return $pages;
 	}
 
+	/**
+	 * Override of the same function in order to support ES6 files
+	 * Duplicate of https://gerrit.wikimedia.org/g/mediawiki/core/+/6fd9245f4ce47a77dc76f70994952cd6da2d1db7/includes/ResourceLoader/Module.php#1083
+	 * Can be removed when moving to a MW version >= 1.4
+	 * @param string $fileName
+	 * @param string $contents
+	 * @return string
+	 */
+	protected function validateScriptFile( $fileName, $contents ) {
+		if ( !$this->getConfig()->get( MainConfigNames::ResourceLoaderValidateJS ) ) {
+			return $contents;
+		}
+		$cache = MediaWikiServices::getInstance()->getMainWANObjectCache();
+		// Cache potentially slow parsing of JavaScript code during the
+		// critical path. This happens lazily when responding to requests
+		// for modules=site, modules=user, and Gadgets.
+		$error = $cache->getWithSetCallback(
+			$cache->makeKey(
+				'resourceloader-userjsparse',
+				self::USERJSPARSE_CACHE_VERSION,
+				md5( $contents ),
+				$fileName
+			),
+			$cache::TTL_WEEK,
+			static function () use ( $contents, $fileName ) {
+				try {
+					Peast::ES2016( $contents )->parse();
```

Comment:
I know that Laura wanted ES2020 is possible

Reply:
Fine with me, 2020 should be "old enough" to have decent coverage